truffle/contracts/VaultNotary/VaultNotary.sol

```diff
@@ -0,0 +1,243 @@
+      * @param addressToGrant address The address to grant permission
+      */
+    function grantUpdateHashPermission(bytes16 vaultId, address addressToGrant)
+        external
+        vaultOwnerOnly(vaultId)
+        isRegistered(vaultId)
+    {
+        notaryMapping[vaultId].hashAcl[addressToGrant] = true;
+        emit UpdateHashPermissionGranted(msg.sender, vaultId, addressToGrant);
+    }
+
+    /** @notice Function to revoke update permission to both the Hash field in one vault
+      * @param vaultId The ID of Vault to revoke permission
+      * @param addressToRevoke address The address to revoke permission
+      */
+    function revokeUpdateHashPermission(bytes16 vaultId, address addressToRevoke)
+        external
+        vaultOwnerOnly(vaultId)
+        isRegistered(vaultId)
+    {
+        notaryMapping[vaultId].hashAcl[addressToRevoke] = false;
+        emit UpdateHashPermissionRevoked(msg.sender, vaultId, addressToRevoke);
+    }
+
+    /** @notice Function to grant update permission to both Uri field in one vault
+      * @param vaultId bytes16 The ID of Vault to grant permission
+      * @param addressToGrant address The address to grant permission
+      */
+    function grantUpdateUriPermission(bytes16 vaultId, address addressToGrant)
+        external
+        vaultOwnerOnly(vaultId)
+        isRegistered(vaultId)
+    {
+        notaryMapping[vaultId].uriAcl[addressToGrant] = true;
+        emit UpdateUriPermissionGranted(msg.sender, vaultId, addressToGrant);
+    }
+
+    /** @notice Function to revoke update permission to both the Uri field in one vault
+      * @param vaultId The ID of Vault to revoke permission
+      * @param addressToRevoke address The address to revoke permission
+      */
+    function revokeUpdateUriPermission(bytes16 vaultId, address addressToRevoke)
+        external
+        vaultOwnerOnly(vaultId)
+        isRegistered(vaultId)
+    {
+        notaryMapping[vaultId].uriAcl[addressToRevoke] = false;
+        emit UpdateUriPermissionRevoked(msg.sender, vaultId, addressToRevoke);
+    }
+
+    /** @notice This is used in unit tests to verify the values are correct after using the setters
+      * Can also be used for outside users to read the vaultUri and vaultHash
+      * @param vaultId bytes16 The ID of the Vault to query
+      * @return vaultUri string The uri of the vault
+      * @return vaultHash string The hash of the vault
+      */
+    function getVaultNotaryDetails(bytes16 vaultId)
+        external
+        view
+        returns(string memory vaultUri, string memory vaultHash)
+    {
+        vaultUri = notaryMapping[vaultId].vaultUri;
+        vaultHash = notaryMapping[vaultId].vaultHash;
+    }
+
+    /** @notice Function to set the vault URI
+      * @param vaultId bytes16 ID of the vault to set
+      * @param vaultUri string The vault URI to set
+      */
+    function setVaultUri(bytes16 vaultId, string memory vaultUri)
+        public
+        canUpdateUri(vaultId)
```

Comment:
This should include the `isRegistered` modifier.  Currently if this function is called with an unregistered `vaultId`, the rejection message will be `Only the vault owner or whitelisted users can update vault URI` when it would be more correct to reject with `Vault ID does not exist`.